strings/base_string.h

```diff
@@ -550,10 +550,38 @@ namespace winrt::impl
         auto end = std::copy(std::begin(temp), result.ptr, buffer);
         return hstring{ std::wstring_view{ buffer, static_cast<std::size_t>(end - buffer)} };
     }
+
+#if __cpp_lib_format >= 202207L
```

Comment:
This is the updated feature testing macro's value, I personally grabbed it directly from [the PR to the STL repo](https://github.com/microsoft/STL/pull/3074) but we can also look at [the standard draft](http://eel.is/c++draft/version.syn#lib:__cpp_lib_format)

Reply:
Is there any reason to compare its value instead of just `#ifdef __cpp_lib_format`? (The existing code does this with __cpp_lib_coroutine for example.)

Reply:
Because std::format_string is a later addition, it's possible that __cpp_lib_format is defined but std::format_string is not available

Reply:
And here I thought feature test macros meant I no longer need to embed version numbers in source code. 😜